lib/puppet/indirector/resource_type/parser.rb

```diff
@@ -100,4 +100,9 @@ def resource_types_in(environment)
     environment.check_for_reparse
     environment.known_resource_types
   end
+
+  def allow_remote_requests?
+    Puppet.deprecation_warning("The resource_type endpoint is deprecated in favor of the environment_classes endpoint. See https://github.com/puppetlabs/puppet-server/blob/master/documentation/puppet-api/v3/environment_classes.md")
```

Comment:
I don't know what our usual policy is on links within the code. Is there a docs.puppet.com link we could point to, or is pointing to the source code the best? If so, is it master we want to point to (won't that always be the documentation for an upcoming and unreleased version)?